cfpb/cfgov-refresh
# Add Turbolinks JS library behind a feature flag

Turbolinks speeds up page loads by fetching and caching content. Is it a good idea? Who knows. We've been experimenting with it on eRegs and it might be appropriate for the larger cf.gov ecosystem. This PR will enable turbolinks across cf.gov when the TURBOLINKS feature flag is enabled.

https://github.com/turbolinks/turbolinks

## Additions

- Turbolinks on-demand JS file.
- `TURBOLINKS` feature flag that loads the above file on cf.gov pages when enabled.

## Testing

1. `./frontend.sh`
1. Enable the `TURBOLINKS` feature flag.
1. Load http://localhost:8000/ and click around to a few random pages. You should see a new progress bar at the top of the screen. Subsequent visits to the same pages should be instantaneous.

## Notes

- I am not endorsing the use of turbolinks. It's kind of scary and would require a lot of testing to ensure we're not breaking any functionality. But it's pretty neat and worth checking out.
